Add gui to get loadout

AdlerSalbei opened the pull request:
please test

McDiod commented:
FTR, this is what it currently looks like:

![20190907153941_1](https://user-images.githubusercontent.com/16549968/64475697-336b3e80-d186-11e9-91c8-fdbf05d2e099.jpg)

Review:
* window content looks right-heavy
* "Name" should be "Faction Name" or something
* export should have a default name value, if user doesn't enter one
* name value should filter for what is actually allowed as a classname (e.g. no spaces)
* explanatory tooltips missing
* "SQL Class" and "PTL Class" seem to be wrong? Should be "Squad Medic Class" and "Platoon Medic Class" or an abbreviation of those
* Why select a class for the medic roles at all? Why not make it rank dependent as we do in grad-factions?
* no idea what "Default Items" checkbox does

AdlerSalbei commented:
>     * window content looks right-heavy

Same style as Arma uses.

>     * "Name" should be "Faction Name" or something
Choose Loadout Name, becouse it's not a faction
 
>     * export should have a default name value, if user doesn't enter one
Added
>     * name value should filter for what is actually allowed as a classname (e.g. no spaces)
Also added
>     * explanatory tooltips missing
Also added
>     * "SQL Class" and "PTL Class" seem to be wrong? Should be "Squad Medic Class" and "Platoon Medic Class" or an abbreviation of those
Changed
>     * Why select a class for the medic roles at all? Why not make it rank dependent as we do in grad-factions?
Becous A) it's shit and B) it creates less problems 
>     * no idea what "Default Items" checkbox does
See Tooltip 😉

McDiod commented:
* die anderen Fenster (z.B. Attributes >> General) sehen nicht so rechtslastig aus
* Tooltips sind normalerweise auf dem Text und nicht auf der Auswahlcontrol
* systemChat funktioniert im Editor nicht
* https://community.bistudio.com/wiki/BIS_fnc_filterString
* wenn du freundlicherweise dein Zeug testen würdest, bevor es jemand reviewen soll (siehe unten)

```
17:22:39 Error in expression <uredText pushBack format ["class %1 {", ];
_structuredText pushBack (_tab + "cla>
17:22:39   Error position: <];
_structuredText pushBack (_tab + "cla>
17:22:39   Error Missing [
17:22:39 File x\GRAD\addons\getLoadoutFromUnits\functions\fnc_onBtnOk.sqf..., line 57
```

AdlerSalbei commented:
>     * die anderen Fenster (z.B. Attributes >> General) sehen nicht so rechtslastig aus
Hab die länge gekürzt
>     * Tooltips sind normalerweise auf dem Text und nicht auf der Auswahlcontrol
Stimmt, hab ich angepsst
>     * systemChat funktioniert im Editor nicht
Doch, siehe Bild.
>     * https://community.bistudio.com/wiki/BIS_fnc_filterString
Okay, damit ist die Debug meldung aber weg.
>     * wenn du freundlicherweise dein Zeug testen würdest, bevor es jemand reviewen soll (siehe unten)
Sry, den hab ich verpasst

![20190908234938_1](https://user-images.githubusercontent.com/12427159/64494962-dbbaf900-d293-11e9-9a70-c71ce36c271b.jpg)

nomisum commented:
Was wird man mit dem UI machen können? (im ggs zum bisherigen Export)

AdlerSalbei commented:
Durch die GUI kannst du direkt in Arma einen Namen vergeben, du kannst angeben das bestimmte Klassen einen der Medic Stufen sind. Diese erhalten dann in ihr Inventar die Medic Sachen. 
Desweiteren kann man durch das anhaken von der default Items Option, die standart items die wir normalerweise mit haben in die uniform adden. Also Map Tools, Medic Items, Flashlights, etc.

McDiod commented on addons/getLoadoutsFromUnits/functions/fnc_addMedicItems.sqf:
```diff
@@ -6,103 +6,126 @@ private _log = [];
 private _items = [];
 private _loadout = getUnitLoadout _unit;
 
+if (isNil "_loadout" || {_loadout isEqualTo []}) exitWith {diag_log "GRAD_Mod getLoadouts from Unit: Loadout is Nil!"};
```
```suggestion
if (isNil "_loadout" || {_loadout isEqualTo []}) exitWith {ERROR("Loadout is nil!")};
```

McDiod commented on addons/getLoadoutsFromUnits/functions/fnc_addMedicItems.sqf:
```diff
@@ -6,103 +6,126 @@ private _log = [];
-   }forEach _items;
+    {
+        systemChat "GRAD_getLoadoutsFromUnits: Items could not be added to the Loadout. See RPT";
+        diag_log format ["GRAD_getLoadoutsFromUnits: Could not add Items: %1 of Amount: %2 to Unit: %3", _x select 0, _x select 1, _unit];
```
```suggestion
        ERROR_3("Could not add items: %1 of amount: %2 to unit: %3", _x select 0, _x select 1, _unit);
```

McDiod commented on addons/getLoadoutsFromUnits/functions/fnc_addStandardItems.sqf:
```diff
@@ -0,0 +1,29 @@
+
+private _loadout = getUnitLoadout _unit;
+
+if (isNil "_loadout" || {_loadout isEqualTo []}) exitWith {diag_log "GRAD_Mod getLoadouts from Unit: Loadout is Nil!"};
```
```suggestion
if (isNil "_loadout" || {_loadout isEqualTo []}) exitWith {ERROR("Loadout is nil!")};
```

McDiod commented:
Unsicher was jetzt conflicted, wurde doch eigentlich nichts in den master gemerged seitdem oO
